web-components/src/components/form/CreditRetireForm.tsx

```diff
@@ -90,41 +94,72 @@ const sxs = {
 
 export interface BottomCreditRetireFieldsProps {
   mapboxToken: string;
+  arrayPrefix?: string;
+  arrayIndex?: number;
 }
 
+type LocationType = {
+  country?: string;
+  stateProvince?: string;
+  postalCode?: string;
+};
+
 export const BottomCreditRetireFields: React.FC<BottomCreditRetireFieldsProps> =
-  ({ mapboxToken }) => {
+  ({ mapboxToken, arrayPrefix = '', arrayIndex }) => {
     const styles = useStyles();
-    const {
-      values: { country, stateProvince, postalCode },
-      setFieldValue,
-    } = useFormikContext<RetireFormValues>();
+    const { values, setFieldValue } = useFormikContext<
+      RetireFormValues | RetireFormValuesArray
+    >();
+
+    const item =
+      typeof arrayIndex === 'number'
+        ? (values as RetireFormValuesArray).recipients[arrayIndex]
+        : (values as RetireFormValues);
+
+    const { country, stateProvince, postalCode } = item;
 
     useEffect(() => {
-      const setRetirementLocation = async (): Promise<void> => {
-        const isoString = await getISOString(mapboxToken, {
-          countryKey: country,
-          stateProvince,
-          postalCode,
-        });
-        setFieldValue('retirementLocation', isoString);
+      const retirementLocationName = `${arrayPrefix}retirementLocation`;
+
+      const setRetirementLocation = async ({
+        country,
+        stateProvince,
+        postalCode,
+      }: LocationType): Promise<void> => {
+        try {
+          const isoString = await getISOString(mapboxToken, {
+            countryKey: country,
+            stateProvince,
+            postalCode,
+          });
+          setFieldValue(retirementLocationName, isoString);
+        } catch (err) {
+          // TODO
```

Comment:
As we talked about, as it is the calculation of a data derived from inputs, which accesses an external service that can fail, instead of associating the error to a field or to the form itself, we have chosen to display the error banner to report that the geocoding service is not available.